### PR TITLE
Maryland 100 year subtraction

### LINKED ISSUE #2818: Maryland 100 year subtraction
References: 
https://www.marylandtaxes.gov/forms/22_forms/Resident_Booklet.pdf
https://mgaleg.maryland.gov/2023RS/fnotes/bil_0001/sb0461.pdf

### PR DESCRIPTION
**Pull request by SirMalamute**
Fixes #2818

### Summary
📝📱🌐

Add a new policy parameter `child_allowance` to the US tax and benefit system. Update the `household_net_income` function to include the child allowance in the calculation. Add tests and documentation for the new parameter and function.

> _`is_valid` checks_
> _refactored for clarity_
> _autumn leaves falling_

### Walkthrough
*  Add a new parameter `child_allowance.amount` to the `parameters.yaml` file, which sets the monthly amount of the child allowance benefit per child ()

### CODE REVIEW

File: policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/age_threshold.yaml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+description: Maryland allows filers at or above this age to receive the hundred year subtraction.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/age_threshold.yaml:1]:
```suggestion
description: Maryland allows filers this age or older to receive the hundred year subtraction. 
```

File: policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/amount.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+description: MD hundred year subtraction amount
+values:
+  2021-01-01: 100_000
+metadata:
+  reference:
+    - title: MD 2022 Resident Booklet
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/amount.yaml:6]:
```suggestion
    - title: MD 2022 Resident Booklet (ya)
```

File: policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/amount.yaml [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+      href: https://www.marylandtaxes.gov/forms/22_forms/Resident_Booklet.pdf#page=12
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/amount.yaml:7]:
```suggestion
      href: https://www.marylandtaxes.gov/forms/22_forms/Resident_Booklet.pdf#page=18
```

**Review by MaxGhenis — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/age_threshold.yaml [hunk 4]
```diff
@@ -0,0 +1,9 @@
+description: Maryland allows filers at or above this age to receive the hundred year subtraction. 
+values:
+  2022-01-01: 100
+metadata:
+  reference:
+    - title: MD 2022 Resident Booklet
+      href: https://www.marylandtaxes.gov/forms/22_forms/Resident_Booklet.pdf#page=12
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/age_threshold.yaml:7]: https://casetext.com/statute/code-of-maryland/article-tax-general/title-10-income-tax/subtitle-2-maryland-taxable-income-calculations-for-individual/part-ii-maryland-adjusted-gross-income/section-10-207-subtractions-from-federal-adjusted-gross-income-in-general (nn)

File: policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/amount.yaml [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+  unit: currency-USD
+  period: year
+  label: MD subtraction agi 100 year amount
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/amount.yaml:10]:
```suggestion
  label: Maryland hundred year subtraction amount
```

File: policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/age_threshold.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/age_threshold.yaml:1]: need a changelog_entry file

File: policyengine_us/variables/gov/states/md/tax/income/agi/subtractions/hundred_year/md_hundred_year_exemption_eligible.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/md/tax/income/agi/subtractions/hundred_year/md_hundred_year_exemption_eligible.py:1]: need a unit test

File: policyengine_us/variables/gov/states/md/tax/income/agi/subtractions/hundred_year/md_hundred_year_exemption_eligible.py [hunk 6]
```diff
@@ -0,0 +1,15 @@
+from policyengine_us.model_api import *
+
+
+class md_hundred_year_exemption_eligible(Variable):
+    value_type = bool
+    entity = Person
+    label = "Maryland hundred year exemption eligibility"
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/md/tax/income/agi/subtractions/hundred_year/md_hundred_year_exemption_eligible.py:7]:
```suggestion
    label = "Eligible for the Maryland hundred year exemption"
```

File: policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/amount.yaml [hunk 7]
```diff
@@ -0,0 +1,11 @@
+description: Maryland provides this hundred year subtraction amount.
+values:
+  2021-01-01: 0
+  2022-01-01: 100_000
+metadata:
+  reference:
+    - title: MD 2022 Resident Booklet (ya)
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/amount.yaml:7]: is there a legal code or bill that we can add?

File: policyengine_us/tests/policy/baseline/gov/states/md/tax/income/agi/subtractions/hundred_year/md_hundred_year_exemption_eligible.py
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/md/tax/income/agi/subtractions/hundred_year/md_hundred_year_exemption_eligible.py:1]: needs to be a yaml file

File: policyengine_us/variables/gov/states/md/tax/income/agi/subtractions/hundred_year/md_hundred_year_exemption.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/md/tax/income/agi/subtractions/hundred_year/md_hundred_year_exemption.py:1]: need a unit test with eligibility as an input

Comment (SirMalamute) [reply policyengine_us/variables/gov/states/md/tax/income/agi/subtractions/hundred_year/md_hundred_year_exemption.py:1]: i did that i think

File: policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/amount.yaml (hunk 7 above)
Comment (SirMalamute) [reply policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/amount.yaml:7]: will add

File: changelog_entry.yaml
Comment (SirMalamute) [inline changelog_entry.yaml:1]: Maryland hundred year subtraction.

File: policyengine_us/variables/gov/states/md/tax/income/agi/subtractions/hundred_year/md_hundred_year_exemption.py
Comment (SirMalamute) [inline policyengine_us/variables/gov/states/md/tax/income/agi/subtractions/hundred_year/md_hundred_year_exemption.py:1]: change all to subtraction

File: policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/amount.yaml
Comment (SirMalamute) [inline policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/amount.yaml:1]: no quotation marks around href

File: policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/age_threshold.yaml [hunk 8]
```diff
@@ -0,0 +1,12 @@
+description: Maryland qualifies filers for the hundred year subtraction at or above this age threshold.
+values:
+  2021-01-01: 0
+  2022-01-01: 100
+metadata:
+  reference:
+    - title: MD 2022 Resident Booklet (ya)
+      href: https://www.marylandtaxes.gov/forms/22_forms/Resident_Booklet.pdf#page=18
+    - title: House Bill 186
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/age_threshold.yaml:9]: add section of the bill

Comment (PavelMakarchuk) [reply policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/age_threshold.yaml:9]: last comment

Comment (SirMalamute) [reply policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/age_threshold.yaml:9]: i'm unsure on what you mean

File: policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/age_threshold.yaml [hunk 9]
```diff
@@ -0,0 +1,12 @@
+description: Maryland qualifies filers for the hundred year subtraction at or above this age threshold.
+values:
+  2021-01-01: 0
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/age_threshold.yaml:3]:
```suggestion
```

File: policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/amount.yaml [hunk 10]
```diff
@@ -0,0 +1,13 @@
+description: Maryland provides this hundred year subtraction amount.
+values:
+  2021-01-01: 0
```
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/amount.yaml:3]:
```suggestion
```

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: add to subtraction sources to hook it up to the net income tree

**Review by PavelMakarchuk — APPROVED** [review]

File: policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/amount.yaml [hunk 11]
```diff
@@ -0,0 +1,12 @@
+description: Maryland provides this hundred year subtraction amount.
+values:
+  2022-01-01: 100_000
+metadata:
+  reference:
+    - title: MD 2022 Resident Booklet (ya)
+      href: https://www.marylandtaxes.gov/forms/22_forms/Resident_Booklet.pdf#page=18
+    - title: House Bill 186
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/amount.yaml:8]:
```suggestion
    - title: Md. Code, Tax-Gen. § 10-207(nn)
      href: https://casetext.com/statute/code-of-maryland/article-tax-general/title-10-income-tax/subtitle-2-maryland-taxable-income-calculations-for-individual/part-ii-maryland-adjusted-gross-income/section-10-207-subtractions-from-federal-adjusted-gross-income-in-general
    - title: House Bill 186
```

File: policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/amount.yaml [hunk 12]
```diff
@@ -0,0 +1,12 @@
+description: Maryland provides this hundred year subtraction amount.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/amount.yaml:1]:
```suggestion
description: Maryland subtracts this amount from adjusted gross income under the hundred year subtraction.
```

File: policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/age_threshold.yaml [hunk 13]
```diff
@@ -0,0 +1,11 @@
+description: Maryland qualifies filers for the hundred year subtraction at or above this age threshold.
```
Comment (MaxGhenis) [inline policyengine_us/parameters/gov/states/md/tax/income/agi/subtractions/hundred_year/age_threshold.yaml:1]:
```suggestion
description: Maryland limits the hundred year subtraction to individuals this age or older.
```

**Review by MaxGhenis — APPROVED** [review]
good to merge after this

**Pull request merged**